feat: Created react-navigation guide (#1966)
callstack/react-native-paper: docs/pages/8.react-navigation.md

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen name="Home" component={HomeScreen} >
        <Stack.Screen name="Details" component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
```

Now our navigation stack contains two screens and will render `HomeScreen` or `DetailsScreen` components according to current navigation state. But we have not implemented those components yet. Let's do this then:

```js
import React from 'react';
import {View, Text, Button, StyleSheet} from 'react-native';

function HomeScreen() {
  return (
    <View style={style.container}>
      <Text>Home Screen</Text>
    </View>
  );
}

function DetailsScreen() {
  return (
    <View style={style.container}>
      <Text>Details Screen</Text>
    </View>
  );
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```

> Once we have that we can start our app and see what we have.

Reviewer: ```suggestion
Once we have that we can run the app and check how Stack looks like.
```

![]('../../../assets/gallery/appbar-dark.png')

To navigate from `HomeScreen` to `DetailsScreen` we can use navigation object provided by `Stack.Screen` component. Every component rendered by `Stack.Screen` have access to navigation object via props. Let's modify our `HomeScreen` component:

```js
function HomeScreen({ navigation }) {
  return (
    <View style={style.container}>
      <Text>Home Screen</Text>
      <Button
        title="Go to details"
        onPress={() => navigation.navigate('Details')}
      />
    </View>
  );
}
```

Our result:

//gif

Now we can navigate between two screens, but we are not done yet.

### Adding AppBar

> We can customize `react-nativation` top bar passing other components:

Reviewer: ```suggestion
We can customize Stack's header by passing custom component:
```

function CustomNavigationBar() {
  return (
    <Appbar.Header>
      <Appbar.Content title="My awesome app" />
    </Appbar.Header>
  );
}
```

We want to see application title on main screen and back arrow button on other screens. We can pass navigation props to our `CustomNavigationBar` component. That way we can know if we are on main screen or not.

First let's pass navigation props to `CustomNavigationBar`:

```js
export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="Home"
        screenOptions={{
          header: (props) => <CustomNavigationBar {...props} />,
        }}>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Details" component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
```

Now we can check if navigation have previous state. If it has we can render back arrow button, otherwise we will render app name as before.

```js
function CustomNavigationBar({ navigation, previous }) {
  return (
    <Appbar.Header>
      {previous ? (
        <Appbar.BackAction onPress={navigation.goBack} />
      ) : (

>         <Appbar.Content title="My awesome app" />

Reviewer: I believe Content should be always visible. It doesn't look good when the header has only back button

      )}
    </Appbar.Header>
  );
}
```

![]('../assets/screenshots/react-nativation-appBar3.gif')

> And that's we need. Now you can use Material design `AppBar` with `react-navigation` to ensure your app works and looks well.

Reviewer: ```suggestion
And that's what we need. Now you can use Material design `AppBar` with `react-navigation` to ensure your app works and looks well.
```
